pkgs/development/libraries/xdg-desktop-portal-gtk/default.nix

```diff
@@ -36,8 +36,16 @@ stdenv.mkDerivation (finalAttrs: {
     gtk3
     xdg-desktop-portal
     gsettings-desktop-schemas # settings exposed by settings portal
-    gnome-desktop
-    gnome-settings-daemon # schemas needed for settings api (mostly useless now that fonts were moved to g-d-s, just mouse and xsettings)
+     # schemas needed for settings api (mostly useless now that fonts were moved to g-d-s, just mouse and xsettings)
+    (runCommand "gnome-settings-daemon-${gnome-settings-daemon.version}-gsettings-schemas" {} ''
+      mkdir -p $out/share
+      cp -r ${gnome-settings-daemon}/share/gsettings-schemas/ $out/share/
+    '')
+  ];
+
+  mesonFlags = [
+    # only useful on Gnome which uses xdg-desktop-portal-gnome
```

Comment:
So [`x-d-p-gtk`](https://github.com/flatpak/xdg-desktop-portal-gtk/blob/37a3c57d6ec9a6a11dc09442b821d95dae9efbf1/src/wallpaper.c#L119) does the same thing as [`x-d-p-gnome`](https://github.com/GNOME/xdg-desktop-portal-gnome/blob/6283cedc4cac52d857d3d380bf2dcd7003ede4f6/src/wallpaper.c#L122) – it sets `org.gnome.desktop.background` GSettings schema from [`g-d-s`](https://gitlab.gnome.org/GNOME/gsettings-desktop-schemas/-/blob/7c4a4a0ef5d21ffe1e6791808ad7be346f49f551/schemas/org.gnome.desktop.background.gschema.xml.in). Looking at [Debian code search](https://codesearch.debian.net/search?q=org.gnome.desktop.background&literal=1), at least the following desktops packages appear to use the schema as consumers:

- ❗ [`budgie-desktop`](https://github.com/BuddiesOfBudgie/budgie-desktop/blob/98a6bfb083b0928fed331764de1c49dbae099816/src/wm/background.vala#L64) – explicitly wants to use [`x-d-p-gtk`](https://github.com/BuddiesOfBudgie/budgie-desktop/blob/98a6bfb083b0928fed331764de1c49dbae099816/data/budgie-portals.conf#L3) for all portals
- [`cinnamon`](https://github.com/linuxmint/cinnamon/blob/520b4db1f0af12c598534efe91f211786a7b2443/js/ui/backgroundManager.js#L12) – it has [its own wallpaper portal](https://github.com/linuxmint/xdg-desktop-portal-xapp/blob/b364687b8a696c78f2d3c46e0c45c2dc28e79a33/src/wallpaper.c#L71)
- [`gnome-flashback`](https://github.com/GNOME/gnome-flashback/blob/baf2a1572d0b30aefaebbfd863c1697ac418d19f/gnome-flashback/libdesktop/gf-background.c#L290) – will use [`x-d-p-gnome`](https://github.com/GNOME/gnome-flashback/blob/baf2a1572d0b30aefaebbfd863c1697ac418d19f/data/xsessions/gnome-flashback-metacity.desktop.in.in#L7)
- [`gnome-kiosk`](https://github.com/GNOME/gnome-kiosk/blob/bb07818bb93d8f14ceb1f7e6b5e5f15a6f031578/compositor/kiosk-backgrounds.c#L26) – will use [`x-d-p-gnome`](https://github.com/GNOME/gnome-kiosk/blob/bb07818bb93d8f14ceb1f7e6b5e5f15a6f031578/kiosk-script/xsessions/gnome-kiosk-script-xorg.desktop.in#L7), though probably not relevant as it is a kiosk DE
- ❗ [`lomiri`](https://gitlab.com/ubports/development/core/lomiri/-/blob/9c030981261e7159641cbd61e600bf3ea25f1f4e/qml/Shell.qml#L196) – uses [`XDG_CURRENT_DESKTOP=Lomiri`](https://gitlab.com/ubports/development/core/lomiri-session/-/blob/cc36885fd44c33b955fbc4eb483476c44c68d442/desktop/lomiri.desktop#L6) and does not appear to have its own portal. Though it might [stop using the setting](https://gitlab.com/ubports/development/core/lomiri/-/merge_requests/138) in the future
- [`phosh`](https://gitlab.gnome.org/World/Phosh/phosh/-/blob/main/src/background-manager.c#L310) – will use [`x-d-p-gnome`](https://gitlab.gnome.org/World/Phosh/phosh/-/blob/main/data/wayland-sessions/phosh.desktop?ref_type=heads#L7)
- [`ukui`](https://github.com/ukui/ukui-window-switch/blob/8a52782d3cc42c459e7d6942d0022c20770bae33/src/ukws_workspace_multiview.cpp#L187), not packaged, possibly only using the schema [on Ubuntu](https://github.com/ukui/ukui-screensaver/blob/8c3e221edb7003b2102fe6ad4d12392520f061bd/src/configuration.cpp#L81-L83)

So removing gnome-desktop breaks at least wallpaper portal in Budgie.

Reply:
maybe we can just have a `-minimal` and `-full` version of this?